discover.go

```diff
@@ -24,5 +21,27 @@ func Discover(glob, dir string) ([]string, error) {
 		}
 	}
 
-	return filepath.Glob(filepath.Join(dir, glob))
+	ls, err := filepath.Glob(filepath.Join(dir, glob))
+	if err != nil {
+		return nil, err
+	}
+	var plugins []string
+
+	// Check for valid plugins files in glob matches
+	for _, f := range ls {
+		stats, err := os.Stat(f)
+		if err != nil {
+			return nil, err
+		}
+		// Skip directories
+		if stats.IsDir() {
+			continue
+		}
+		// If file is executable add to plugins
+		if stats.Mode()&0111 != 0 {
+			plugins = append(plugins, f)
+		}
```

Comment:
It'd be helpful to have the reverse behavior logged, like `[INFO] Skipping non-executable plugin foo`